libs/langchain/langchain/vectorstores/qdrant.py

```diff
@@ -712,6 +716,25 @@ def max_marginal_relevance_search(
                         of diversity among the results with 0 corresponding
                         to maximum diversity and 1 to minimum diversity.
                         Defaults to 0.5.
+            filter: Filter by metadata. Defaults to None.
+            search_params: Additional search params
+            score_threshold:
+                Define a minimal score threshold for the result.
+                If defined, less similar results will not be returned.
+                Score of the returned result might be higher or smaller than the
+                threshold depending on the Distance function used.
+                E.g. for cosine similarity only higher scores will be returned.
+            consistency:
+                Read consistency of the search. Defines how many replicas should be
+                queried before returning the result.
+                Values:
+                - int - number of replicas to query, values should present in all
+                        queried replicas
+                - 'majority' - query all replicas, but return values present in the
+                               majority of replicas
+                - 'quorum' - query the majority of replicas, return values present in
+                             all of them
+                - 'all' - query all replicas, and return values present in all replicas
         Returns:
             List of Documents selected by maximal marginal relevance.
         """
```

Comment:
don't we need to pass all the new params through in line 720

Reply:
Thanks for looking over this! Those are possible values for the consistency flag, not additional parameters.

Reply:
i meant shouldn't L720 (in original code)

```python
        return self.max_marginal_relevance_search_by_vector(
            query_embedding, k, fetch_k, lambda_mult, **kwargs
        )
```

be

```python
        return self.max_marginal_relevance_search_by_vector(
            query_embedding, k, fetch_k, lambda_mult, filter=filter, search_params=search_params, score_threshold=score_threshold, consistency=consistency, **kwargs
        )
```

and then these params also need to be added to signature of max_marginal_relevance_search_by_vector (L813 in original code)